code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,329 @@
+	RegisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT, .proc/disable_drone_creation)
+
+/obj/machinery/computer/camera_advanced/remote_fob/proc/disable_drone_creation()
+	drone_creation_allowed = FALSE
+	UnregisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT)
+
+/obj/machinery/computer/camera_advanced/remote_fob/examine(mob/user)
+	. = ..()
+	var/list/details = list()
+	details +=("It has [metal_remaining] sheets of metal remaining.</br>")
+	details +=("It has [plasteel_remaining] sheets of plasteel remaining.</br>")
+	details +=("It has [sentry_remaining] sentries ready for placement.</br>")
+	to_chat(user, "[details.Join(" ")] ")
+
+/obj/machinery/computer/camera_advanced/remote_fob/give_eye_control(mob/user)
+	. = ..()
+	user.lighting_alpha = 120
+	eyeobj.name = "Remote Construction Drone"
+	if(eyeobj.eye_initialized)
+		eyeobj.setLoc(get_turf(spawn_spot))
+
+
+/obj/machinery/computer/camera_advanced/remote_fob/attack_hand(mob/living/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='warning'>Access Denied!</span>")
+		return
+	if(!drone_creation_allowed)
+		to_chat(user, "<span class='notice'>Communication with the drone impossible due to fuel-residue in deployment zone atmosphere.</span>")
+		return
+	spawn_spot = FALSE
+	switch(tgalert(user, "Summon Drone in:", "FOB Construction Drone Control", "LZ1","LZ2"))
+		if("LZ1")
+			spawn_spot = locate(/obj/docking_port/stationary/marine_dropship/lz1) in SSshuttle.stationary	
+		if("LZ2")
+			spawn_spot = locate(/obj/docking_port/stationary/marine_dropship/lz2) in SSshuttle.stationary
+		else
+			return
+	return ..()
+	
+
+/obj/machinery/computer/camera_advanced/remote_fob/CreateEye()
+	if(spawn_spot)
+		eyeobj = new /mob/camera/aiEye/remote/fobdrone(get_turf(spawn_spot))
+		eyeobj.origin = src
+	return
+
+/obj/machinery/computer/camera_advanced/remote_fob/attackby(obj/item/stack/W, mob/user, params)
+	if(istype(W, /obj/item/stack/voucher/sentry))
+		var/useamount = W.amount
+		sentry_remaining += useamount
+		W.use(useamount)
+		to_chat(user, "<span class='notice'>Sentry voucher redeemed.</span>")
+		playsound(src, 'sound/machines/terminal_insert_disc.ogg', 25, FALSE)
+		return
+	if(istype(W, /obj/item/stack/sheet/metal))
+		if(max_metal >= metal_remaining)
+			to_chat(user, "<span class='notice'>Can't insert any more metal.")
+			return
+		var/useamount = min(W.amount, (max_metal-metal_remaining))
+		metal_remaining += useamount
+		W.use(useamount)
+		to_chat(user, "<span class='notice'>Inserted [useamount] metal sheets.")
+		return
+	if(istype(W, /obj/item/stack/sheet/plasteel))
+		if(max_plasteel >= plasteel_remaining)
+			to_chat(user, "<span class='notice'>Can't insert any more plasteel.")
+			return
+		var/useamount = min(W.amount, (max_plasteel-plasteel_remaining))
+		plasteel_remaining += useamount
+		W.use(useamount)
+		to_chat(user, "<span class='notice'>Inserted [useamount] plateel sheets.")
+		return
+	else
+		to_chat(user, "<span class ='notice'>Invalid material type.</span>")
```

Comment:
The `else` is not needed here, as every `if` properly returns.
Do you want this message to happen always? Do consider if the user is, for example, bashing the machine with a weapon.